### DISCUSSION TITLE
How to retrieve data when both `cdadata` and `cdabackup` are down

### DISCUSSION THREAD

**Discussion by ghost** [opendcs/opendcs#382] [General]
Looks like both `cdadata` and `cdabackup` were taken down for maintenance (https://dcs3.noaa.gov/op_notices/rss.xml).  In such cases, how can OpenDCS continue to retrieve data?

Is there a way to use one of the `DCS` servers?

**Comment by MikeNeilson** [comment]
Those the the public instances provided by NOAA. There are other instances that have satellite reception but it's unlikely they are public. USACE doesn't have any public instances, nor do I think there are plans. USGS has EDDN but I'm not sure who to ask about actually getting access.

That said if you really truly always want to have data, setting up your own DRGS or HRIT is the best bet - or coordinate with an entity that does and is willing to give you access. I'm not sure who else besides Microcom sells the "big commerical, expensive and you're unlikely to have major issues system", pretty sure there's at least one other, but I've been getting data with a combination of:

This: https://www.amazon.com/Nooelec-GOES-Weather-Satellite-Bundle/dp/B08HGQXC7C
and this: https://github.com/pietern/goestools

I need to do a writeup on my whole setup but the goestools software runs does all the data processing and can output the "HRIT files" which is one of the LRGS tabs; pretty sure RoutingSpec could do it directly but I'd have to go verify the config.

the goestools software will run on pretty much any modernish SBC or a laptop; I've got it running on a Raspberry Pi 3b with an LRGS instance that yet-another LRGS connects to to be able to store a larger window of data.

**Reply by ghost** [reply]
@MikeNeilson, Can you provide more details about how to run the goestools? Which command do I need to run, and how to set up configs/input/output?

**Comment by agilmore2** [comment]
As Letecia's email announcing the outage said:

*".... Additionally, the LRGS servers located at WCDAS (CDADATA and
CDABACKUP) will be down during this maintenance period.  The LRGS servers
at NSOF (NLRGS1 and NLRGS2) will remain operational.  Also the LRGS sites
at EDDN in Sioux Falls, SD will be available at lrgseddn1.cr.usgs.gov
<http://lrgseddn1.cr.usgs.gov/>, lrgseddn2.cr.usgs.gov
<http://lrgseddn2.cr.usgs.gov/>, or lrgseddn3.cr.usgs.gov
<http://lrgseddn3.cr.usgs.gov/>...."*



Therefore, the rest of the public LRGS servers are and were available
during the Wallops outage. I strongly recommend using a data source
hotbackup group that includes servers from at least two locations. CDADATA
is usually so busy that I avoid using it as the primary.

Andrew Gilmore

**Reply by ghost** [reply]
I don't know the complete address of `NLRGS1` and `NLRGS2`. I also don't know if my `cdadata` credentials would work there. Do you have additional information on it?

**Comment by agilmore2** [comment]
EDDN LRGS, NLRGS and CDA LRGS all share the same login credentials, so no
further action is needed to gain access to them.

The LRGS status link
[image: image.png]
at https://dcs1.noaa.gov links to
https://dcs1.noaa.gov/lrgs/LrgsSummaryStatus.html which has the information
for the public LRGS:

[image: image.png]

Good luck!

Andrew
